### PR TITLE
Support for launching jobs with GraphBolt enabled (GraphBolt support PR1)

### PR DESCRIPTION
**Pull request by thvasilo**
*Issue #, if available:*

*Description of changes:*

* We introduce support for GraphBolt training and inference, see https://docs.dgl.ai/stochastic_training/index.html for details
* We add a new optional argument `--use-graphbolt` to the configuration. When set to true that sets the dgl.initialize call to include `use_graphbolt=True`, on supported DGL versions.
* For versions of DGL < 2, we print a warning and run regular distributed environment init.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by thvasilo** [comment]
Depends on https://github.com/awslabs/graphstorm/pull/1002

File: python/graphstorm/gsf.py [hunk 1]
```diff
@@ -141,10 +149,32 @@ def initialize(ip_config=None, backend='gloo', local_rank=0, use_wholegraph=Fals
...
+            ip_config,
+            net_type='socket',
+            use_graphbolt=use_graphbolt,
+        )
+    else:
+        if use_graphbolt:
+            logging.warning(
+                (
+                    "use_graphbolt was 'true' but but DGL version was %s. "
+                    "GraphBolt requires DGL version >= 2.x."
```
Comment (classicsong) [inline python/graphstorm/gsf.py:168]: 2.4?

File: python/graphstorm/gsf.py [hunk 2]
```diff
@@ -141,10 +149,32 @@ def initialize(ip_config=None, backend='gloo', local_rank=0, use_wholegraph=Fals
...
+                ),
+                dgl_version
+                )
+        # We need to use socket for communication in DGL 0.8. The tensorpipe backend has a bug.
```
Comment (classicsong) [inline python/graphstorm/gsf.py:172]: Why does the comment mean?

File: python/graphstorm/gsf.py [hunk 3]
```diff
@@ -141,10 +149,32 @@ def initialize(ip_config=None, backend='gloo', local_rank=0, use_wholegraph=Fals
...
+        # This problem will be fixed in the future.
```
Comment (classicsong) [inline python/graphstorm/gsf.py:173]: Who is going to fix the issue? Can you check with the DGL team?

File: python/graphstorm/gsf.py (hunk 2 above)
Comment (thvasilo) [reply python/graphstorm/gsf.py:172]: This was pre-existing I just moved it. If we don't need it any more I'll remove it.

File: python/graphstorm/gsf.py (hunk 3 above)
Comment (thvasilo) [reply python/graphstorm/gsf.py:173]: Same as above, I didn't write this comment, but looking at the DGL code, there's only support for socket anyway, so not much difference.

Comment (classicsong) [reply python/graphstorm/gsf.py:173]: Let's remove it.

**Review by classicsong — APPROVED** [review]

**Pull request merged**